### PR TITLE
feat: Add Card component with theming and atomic design

### PR DESCRIPTION
**Pull request by PavanVaranasi2**
- Added IconEle component in atoms
- Created CommuteRoutes component in molecules
- Developed JobCard component in organisms
- Applied Material-UI theming and ensured font family consistency
- Removed inline styles and used theme for consistent styling
- Ensured proper structure and readability following atomic design principles
- Imported Montserrat font and applied CssBaseline for global styles

Screenshot 2024-06-20 at 11 41 16 AM

### CODE REVIEW

**Comment by saicharan-zemoso** [comment]
I think you have given more padding on the left.

File: public/bike icon.png
Comment (saicharan-zemoso) [inline public/bike icon.png:1]: the folder structure should be public/assets/icons

File: src/components/atoms/IconEle/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+interface IconEleProps {
+    src: string;
+    alt: string;
+    width?: string | number;
+    height?: string | number;
+    padding?: string;
+    gap?: string;
+  }
+  
+  const IconEle: React.FC<IconEleProps> = ({
```
Comment (saicharan-zemoso) [inline src/components/atoms/IconEle/index.tsx]: rename it to Icon that will be more suitable.

File: src/components/atoms/IconEle/index.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    gap = '0',
+  }) => {
+    return (
+      <img
+        src={src}
+        alt={alt}
+        style={{ width, height, padding, gap }}
+      />
+    );
+  };
```
Comment (saicharan-zemoso) [inline src/components/atoms/IconEle/index.tsx]: provide id and class name instead. and use those to style the icons or use styled component as you have used in commuteRoutes file

File: src/components/molecules/CommuteRoutes/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      </CommuteIconBox>
+      <CommuteIconBox className="commuteRouteBus">
+        <IconEle src='./bus icon.png' alt='Bus' width='18px' height='18px' />
+      </CommuteIconBox>
+      <CommuteIconBox className="commuteRouteCar">
+        <IconEle src='./car icon.png' alt='Car' width='20px' height='17px' />
+      </CommuteIconBox>
+      <CommuteIconBox className="commuteRouteTrain">
+        <IconEle src='./train icon.png' alt='Train' width='15px' height='18px' />
+      </CommuteIconBox>
```
Comment (saicharan-zemoso) [inline src/components/molecules/CommuteRoutes/index.tsx]: use styled component or use id and class and write styles in css file. And store the src and alt in a object and use map.

File: src/components/organisms/JobCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          </Box>
+        </Box>
+
+        <Box sx={{ textAlign: 'left', marginBottom: theme.spacing(4) }}>
+          <Typography variant="subtitle1">
+            User Experience Designer
+          </Typography>
+          <Typography variant="caption" sx={{ color: theme.palette.primary.main }}>
+            HP
+          </Typography>
```
Comment (saicharan-zemoso) [inline src/components/organisms/JobCard/index.tsx]: remove inline styling. use styled component or css file to write the styles.

**Comment by saicharan-zemoso** [comment]
remove any unused files

**Review by PavanVaranasi2 — COMMENTED** [review]
fix: Update files based on PR feedback

**Comment by saicharan-zemoso** [comment]
check the color of the time text it should be same as commute route text

File: public/bike icon.png
Comment (saicharan-zemoso) [inline public/bike icon.png:1]: folder structure pubic/assets/icons

File: src/theme/theme.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
+// src/theme.tsx
+import { createTheme } from '@mui/material';
+
+const theme = createTheme({
```
Comment (saicharan-zemoso) [inline src/theme/theme.tsx:3]: check the colors in the figma you can see text/high emphasis that should be included in the palette:{text:{high:color,low:color,medium:color}} check theme in material ui website

**Review by saicharan-zemoso — COMMENTED** [review]
resolve all the comments

**Review by PavanVaranasi2 — COMMENTED** [review]
Fix: Adjusted Folder Structure and incorporated internal styling with styled api

Screenshot 2024-06-20 at 5 08 15 PM

File: public/bike icon.png
Comment (saicharan-zemoso) [inline public/bike icon.png:1]: remove these duplicate icons

File: src/components/atoms/IconEle/index.tsx (hunk 1 above)
Comment (saicharan-zemoso) [inline src/components/atoms/IconEle/index.tsx]: please rename this component name as Icon

File: src/components/atoms/IconEle/index.tsx
Comment (saicharan-zemoso) [inline src/components/atoms/IconEle/index.tsx:1]: delete this if you have created Icon component again.

File: src/components/organisms/JobCard/index.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+          <br/>
+          <Typography variant="caption">
+            Hyderabad, Telangana, India
+          </Typography>
+        </ContentBox>
```
Comment (saicharan-zemoso) [inline src/components/organisms/JobCard/index.tsx]: make this as a molecule

File: src/components/organisms/JobCard/index.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+              Commute routes available:
+            </Typography>
+            <CommuteRoutes />
+          </InfoBox>
+          <TimeBox>
+            <Typography variant="caption" sx={{ color: theme.palette.text.high }}>
+              1hr 36 mins ago
+            </Typography>
+          </TimeBox>
+        </FooterBox>
```
Comment (saicharan-zemoso) [inline src/components/organisms/JobCard/index.tsx]: make it as a molecule

File: src/components/.DS_Store
Comment (saicharan-zemoso) [inline src/components/.DS_Store:1]: remove this file if it is not used.

**Review by saicharan-zemoso — CHANGES_REQUESTED** [review]
resolve all comments

File: public/bike icon.png
Comment (PavanVaranasi2) [reply public/bike icon.png:1]: There are no duplicate Icons, in my local file folder, I'll just have a look once agian in github
Screenshot 2024-06-21 at 2 17 51 PM

File: src/components/atoms/IconEle/index.tsx (hunk 1 above)
Comment (PavanVaranasi2) [reply src/components/atoms/IconEle/index.tsx]: This is completely deleted actually, as you can see in my local repo

**Review by PavanVaranasi2 — COMMENTED** [review]
- Removed unused public assets: bike icon.png, bus icon.png, car icon.png, hp icon.png, train icon.png, triple dot icon.png.
- Deleted src/components/atoms/IconEle/index.tsx as it is no longer needed.
- Added new components for better organization:
  - src/components/molecules/ContentBox/
  - src/components/molecules/FooterBox/
  - src/components/molecules/HeaderBox/
- Modified src/components/organisms/JobCard/index.tsx

Screenshot 2024-06-21 at 4 23 29 PM

**Review by saicharan-zemoso — APPROVED** [review]
I would suggest  going through modular augmentation for theming . and Use css or styled components and store the constants in a constants file and export it from there. 
follow these in the future

**Comment by PavanVaranasi2** [comment]
Ok i'll definitely follow your suggestions from now on, thank you so much

**Pull request merged**